frontend/views/components/CopyToClipboard.vue

```diff
@@ -0,0 +1,107 @@
+<template lang='pug'>
+.c-wrapper(@click='copyToClipboard')
+  input.c-invisible-input(
+    type='text'
+    ref='input'
+    :value='textToCopy'
+  )
+  slot
+  transition(name='fade')
+    i18n.c-tooltip(
+      tag='p'
+      v-if='ephemeral.tooltipActive'
+    ) Copied to clipboard!
+</template>
+
+<script>
+import Tooltip from '@components/Tooltip.vue'
+
+export default {
+  name: 'copyToClipboard',
+  data () {
+    return {
+      ephemeral: {
+        tooltipActive: false
+      }
+    }
+  },
+  components: {
+    Tooltip
+  },
+  props: {
+    textToCopy: {
+      type: String,
+      required: true
+    }
+  },
+  methods: {
+    copyToClipboard () {
+      // if the user is using the device that supports web share API, use it and then skip the other logics below.
+      if (navigator.share) {
+        this.activateWebShare()
+        return
+      }
+
+      this.$refs.input.select()
+      document.execCommand('copy')
+      this.ephemeral.tooltipActive = true
+      setTimeout(() => {
+        this.ephemeral.tooltipActive = false
+      }, 1500)
+    },
+    activateWebShare () {
+      navigator.share({
+        title: this.L('Your invite'),
+        url: this.textToCopy
+      })
+    }
+  }
+}
+</script>
+
+<style lang='scss' scoped>
+@import '../../assets/style/_variables.scss';
+
+.c-wrapper {
+  position: relative;
+
+  .c-invisible-input {
```

Comment:
can't you use input type hidden?

Reply:
it seems like it doesn't work with type='hidden' because inputElement.select() can't make input grab the text in this case. :)